File: app/src/main/java/com/github/kongpf8848/animation/activity/tween/rotate/Rotate_8_Activity.java
package com.github.kongpf8848.animation.activity.tween.rotate;

import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.view.animation.AccelerateInterpolator;
import android.widget.ImageView;

import com.github.kongpf8848.animation.R;
import com.github.kongpf8848.animation.activity.BaseActivity;


import butterknife.BindView;

/**
 * Created by pengf on 2016/12/29.
 * 图片8旋转动画
 */

public class Rotate_8_Activity extends BaseActivity
{
     @BindView(R.id.iv_loading)
     ImageView iv_loading;


    @Override
    protected int getLayoutId() {
        return R.layout.activity_rotate_8;
    }

    @Override
    protected void initData() {
        startAnimation();
    }

    private void startAnimation()
    {
        ObjectAnimator animator = ObjectAnimator.ofFloat(iv_loading, "rotationY", 180);
        animator.setRepeatMode(ValueAnimator.REVERSE);
        animator.setInterpolator(new AccelerateInterpolator());
        animator.setDuration(1000);
        animator.start();

    }
}

File: app/src/main/java/com/github/kongpf8848/animation/activity/tween/rotate/Rotate_8_Activity.kt
package com.github.kongpf8848.animation.activity.tween.rotate

import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import android.os.Bundle
import android.view.View
import android.view.animation.AccelerateInterpolator
import android.widget.ImageView
import butterknife.BindView
import com.github.kongpf8848.animation.R
import com.github.kongpf8848.animation.activity.BaseActivity
import kotlinx.android.synthetic.main.activity_rotate_8.*

class Rotate_8_Activity : BaseActivity() {

    override fun getLayoutId(): Int {
        return R.layout.activity_rotate_8
    }

    override fun onCreateEnd(savedInstanceState: Bundle?) {
        startAnimation()
    }

    private fun startAnimation() {
        ObjectAnimator.ofFloat(iv_loading, View.ROTATION_Y, 180f).apply {
            repeatMode = ValueAnimator.REVERSE
            repeatCount=ValueAnimator.INFINITE
            interpolator = AccelerateInterpolator()
            duration = 1000
        }.start()
    }
}